site/source/tutorials/005_form_widgets/index.md

```diff
@@ -1,26 +1,159 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:5-6 %}
+
+
+{% aside 'Note' %}
+	If you have been working with the same project throughout this series, you might get some errors when you try to import the widgets. If you do, add a new dependency to the `project.json` file: "@dojo/widgets":"2.0.0-alpha.23" and rerun `npm install` to install Dojo 2's standard widget library.
+{% endaside %}
+
+We are importing the `Button` class that will be used to provide the form's submit button and the `TextInput` class that will provide the data-entry fields for the worker data.
+
+Let's use those classes and a few virtual DOM nodes, to add the visual elements of the form:
+
+```ts
+	protected render(): DNode {
+		return v('form', {
+			classes: this.classes(css.workerForm),
+			onsubmit: this._onSubmit
+		}, [
+			v('fieldset', { classes: this.classes(css.nameField) }, [
+				v('legend', { classes: this.classes(css.nameLabel) }, [ 'Name' ]),
+				w(TextInput, {
+					key: 'input1',
+					label: {
+						content: 'First Name',
+						hidden: true
+					},
+					placeholder: 'Given name',
+					type: 'text' as 'text'
+				}),
+				w(TextInput, {
+					key: 'input2',
+					label: {
+						content: 'Last Name',
+						hidden: true
+					},
+					placeholder: 'Family name',
+					type: 'text' as 'text'
+				})
+			]),
+			w(TextInput, {
+				key: 'input3',
+				label: 'Email address',
+				type: 'email'
+			}),
+			w(Button, {
+				content: 'Save',
+				type: 'submit'
+			})
+		]);
+	}
+```
+
+At this point, the user interface for the form is available, but it doesn't do anything since we haven't specified any event handlers. In the [last](../004_user_interactions/) tutorial, we learned how to add event handlers to custom widgets by assigning a method to an event. When using pre-built widgets, however, we pass the handlers as properties. For example, we are going to need a way to handle each text fields `change` event. To do that, we provide the desired handler function as the `onChange` property that is passed to the widget. Update the `render` method once again:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:33-110 %}
+
+This form of the `render` method now does everything that we need it to: it creates the user interface and registers the event handlers that will update the application as the user enters information.
```

Comment:
"This form of the `render` method now does everything that we need it to: it creates the user interface and registers the event handlers that will update the application as the user enters information" -> "This form of the `render` method now does everything that we need from it: creates the user interface and registers the event handlers that will update the application as the user enters information"